link/kprobe.go

```diff
@@ -26,6 +27,59 @@ var (
 	}{}
 )
 
+type probeType uint8
+
+const (
+	kprobeType probeType = iota
+	uprobeType
+)
+
+func (pt probeType) String() string {
```

Comment:
Reading the methods on `probeType` here and below, I find the use of `switch` a bit confusing. Since this type represents a boolean, the following pattern would be easier to parse:
```go
if pt == kprobeType {
  return "kprobe"
}
return "uprobe"
```